### PR TITLE
Highlight the selected parent menu when a submenu is open.

### PR DESCRIPTION
**Pull request by EstherDarkish**

### CODE REVIEW

File: src/components/utils/custom-nested-menu.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+import React, { useState } from 'react';
+import { NestedMenuItem } from 'mui-nested-menu';
+import { Box } from '@mui/system';
+import MenuItem from '@mui/material/MenuItem';
+import { mergeSx } from './functions';
+
+const styles = {
+    highlightedParentLine: (theme) => ({
+        backgroundColor: theme.palette.action.hover,
```
Comment (ghelloumi) [inline src/components/utils/custom-nested-menu.js:15]: you can use "action.hover" without theme.palette

File: src/components/utils/custom-nested-menu.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        color: theme.palette.primary.main,
```
Comment (ghelloumi) [inline src/components/utils/custom-nested-menu.js:16]: idem

File: src/components/utils/custom-nested-menu.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        transition: 'all 300ms ease',
+    }),
+    highlightedLine: (theme) => ({
+        transition: 'all 300ms ease',
+        '&:hover': {
+            backgroundColor: theme.palette.action.hover,
```
Comment (ghelloumi) [inline src/components/utils/custom-nested-menu.js:22]: idem

File: src/components/utils/custom-nested-menu.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            color: theme.palette.primary.main,
```
Comment (ghelloumi) [inline src/components/utils/custom-nested-menu.js:23]: idem

File: src/components/utils/custom-nested-menu.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            transition: 'all 300ms ease',
```
Comment (ghelloumi) [inline src/components/utils/custom-nested-menu.js:24]: you can remove this line, it is not necessary to add this css property two times (line 20) to the same class

File: src/components/utils/custom-nested-menu.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            </Box>
+        </NestedMenuItem>
+    );
+};
+
+export const CustomMenuItem = (props) => {
+    const { sx, children, ...other } = props;
+
+    return (
+        <MenuItem sx={mergeSx(styles.highlightedLine, sx)} {...other}>
```
Comment (ghelloumi) [inline src/components/utils/custom-nested-menu.js:58]: you can remove children from props destruction and then remove {children}

=> `    const { sx, ...other } = props;
<MenuItem sx={mergeSx(styles.highlightedLine, sx)} {...other}/>`

File: src/components/utils/custom-nested-menu.js (hunk 1 above)
Comment (EstherDarkish) [reply src/components/utils/custom-nested-menu.js]: Done

File: src/components/utils/custom-nested-menu.js (hunk 2 above)
Comment (EstherDarkish) [reply src/components/utils/custom-nested-menu.js]: Done

File: src/components/utils/custom-nested-menu.js (hunk 4 above)
Comment (EstherDarkish) [reply src/components/utils/custom-nested-menu.js]: Done

File: src/components/utils/custom-nested-menu.js (hunk 6 above)
Comment (EstherDarkish) [reply src/components/utils/custom-nested-menu.js]: Thanks, that's better.

File: src/components/utils/custom-nested-menu.js (hunk 5 above)
Comment (EstherDarkish) [reply src/components/utils/custom-nested-menu.js]: Yes, done.

File: src/components/utils/custom-nested-menu.js (hunk 3 above)
Comment (EstherDarkish) [reply src/components/utils/custom-nested-menu.js]: Done

**Review by ghelloumi — APPROVED** [review]

**Review by souissimai — APPROVED** [review]
tests are ok , code seems good

**Pull request merged**